src/host/pages/IMUMonitorPage.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+  * Project:  KBox
+  * Purpose:  IMU Monitor Page for HDG, Heel & Pitch (+ Cal planned)
+  * Author:   (c) Ronnie Zeiller, 2017
+  * KBox:     (c) 2017 Thomas Sarlandie thomas@sarlandie.net
+
+  The MIT License
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+#include <stdio.h>
+#include "common/signalk/SKUpdate.h"
+#include "IMUMonitorPage.h"
+#include <KBoxLogging.h>
+#include "signalk/SKUnits.h"
+#include "services/IMUService.h"
+#include "common/signalk/SKUpdateStatic.h"
+#include "common/signalk/SKUnits.h"
+
+
+IMUMonitorPage::IMUMonitorPage(SKHub& hub) {
+  _IMU_HDGFiltered    = 361;
+  _IMU_PitchFiltered  = 181;
+  _IMU_HeelFiltered   = 181;
+
+  // Damping values 1.....100 --> 0.5 no filter ......0.005 max filter
+  // for higher frequencies even more
+  // TODO: make KBox setting for damping values 1.......100
+
+  // auto damping factored by the IMUServiceInterval which is
+  // probably between 50 and 100 (20 - 10 Hz)
+  // TODO: set and get IMUServiceInterval for automatic modus -> #define IMUServiceInterval ??
+
+  _dampingHdg = _IMUServiceInterval / 1400.000; // 0.05
+  if (_dampingHdg>0.5) _dampingHdg=0.5;
+
+
+  // Heel and Pitch need smaller damping
+  _dampingHeelPitch = _IMUServiceInterval / 700.000; // 0.1 .... 0.1;
+  if (_dampingHeelPitch>0.5) _dampingHeelPitch=0.5;
+
+  INFO( "HDG Damping factor taken for IIR-filter: %f", _dampingHdg );
+  INFO( "Heel/Pitch Damping factor taken for IIR-filter: %f", _dampingHeelPitch );
+
+  // set filter type
+  IMU_HdgFilter.setType(IIRFILTER_TYPE_DEG);
```

Comment:
The library I recommended does not support type degree unfortunately. We would need to add this.